src/main/scala/com/microsoft/hyperspace/index/sources/interfaces.scala

```diff
@@ -105,50 +106,52 @@ trait FileBasedSourceProvider extends SourceProvider {
    *
    * If the given logical relation does not belong to this provider, None should be returned.
    *
-   * @param logicalRelation Logical relation to compute signature from.
+   * @param logicalPlan Logical relation to compute signature from.
    * @return Signature computed if the given 'logicalRelation' can be processed by this provider.
    *         Otherwise, None.
    */
-  def signature(logicalRelation: LogicalRelation): Option[String]
+  def signature(logicalPlan: LogicalPlan): Option[String]
 
   /**
-   * Retrieves all input files from the given [[LogicalRelation]].
+   * Retrieves all input files from the given [[LogicalPlan]].
    *
-   * @param logicalRelation Logical relation to retrieve input files from.
+   * @param logicalPlan Logical relation to retrieve input files from.
    * @return List of [[FileStatus]] for the given relation.
    */
-  def allFiles(logicalRelation: LogicalRelation): Option[Seq[FileStatus]]
+  def allFiles(logicalPlan: LogicalPlan): Option[Seq[FileStatus]]
 
   /**
-   * Constructs the basePath for the given [[FileIndex]].
-   *
-   * @param location Partitioned data location.
-   * @return Optional basePath to read the given partitioned location as explained below:
-   *         Some(Some(path)) => The given location is supported and partition is specified.
-   *         Some(None) => The given location is supported but partition is not specified.
-   *         None => The given location is not supported.
+   * Constructs the basePath for the given [[LogicalPlan]].
+   *
+   * @param logicalPlan Logical plan to extract the base path from.
```

Comment:
I see "Logical relation" in other comment. Shall we make this consistent?

Reply:
👍